src/genericactiondepotlib/src/actionWriteSerial.cpp

```diff
@@ -98,7 +98,7 @@ execution ActionWriteSerial::execute(const TestRepetitions& testrepetition)
     }
 
     // Write to serial port
-    unsigned char msg[] = { value_.c_str()[0] };
+    unsigned char msg[] = { static_cast<unsigned char>(value_.c_str()[0]) };
     write(serial_port, msg, sizeof(msg));
     TXLOG(Severity::debug)<<"Set value:"<<value_<<" To:"<<port_<<std::endl;
```

Comment:
I  guess this will send just the first character of `value_`  to the serial  port, not the all content of `value_`, is this expected @triccyx @Nicogene ?

Reply:
Not sure about this fix as @traversaro  notice.

Reply:
To clarify, I think that also the code before the fix was just sending the first character (and not the complete string `value_` ) even  before this fix.

Reply:
Yes I didn't changed the behaviour, I just add the cast because macOs (and probably also windows) complains about it.

Reply:
Ok we can try